### DISCUSSION TITLE
Fast charging Soc / Eco mode Soc   (State Of Charge)

### DISCUSSION THREAD

**Discussion by deveylder** [mletenay/home-assistant-goodwe-inverter#330] [Q&A]
Hallo everyone,

Might sound stupid and it probably is but i don't get the option "Fast charging Soc" or  "Eco mode Soc". I do understand Fast charging Power and Eco mode power.   Not sure  => Is this  a regulator for the amount of power that you select (eco mode power) ex 100% and the other one the maximum of battery percentage it can charge (eco mode soc) ex 50% ?

**Comment by edwardhaase** [comment]
I am struggling at a similar point. Where may i find an explanation what the different Inverter statuses 'General mode', 'Off grid mode', 'Backup mode', 'Eco mode', 'Peak shaving mode', 'Eco charge mode' and 'Eco discharge mode' exactly stand for. This also in respect of input values as of 'Eco mode power', 'Depth of discharge', 'Eco mode SoC'. I do have only some guesses but nothing tangible.

Is there anybody who may point me to some more information about what to set to charge/discharge the battery to certain levels and how to force using only battery or only grid (without usage of battery at that time).

Many thanks in advance!

**Reply by edwardhaase** [reply]
How to configure primary use of power from grid? A kind of opposite of 'Peak shaving mode'...load battery up to 100% and use primarily the power from grid and not from battery at cheap price situation maybe described as 'Grid parallel mode'?

**Comment by MrBamse** [comment]
I've struggled repeatedly with finding the same informarion. As great as this repo is the documentation needa some work and I'd be happy to help. I just need to find some sources explaining the functions.

As for most of the "work modes" I have assumed that they are the exact counter part of the work mode with the same name on the inverter (information on which can be found in the official GoodWe documentation of your inverter).

The Eco-modes though are a different story. I believe I once found a thread here in the "discussions" referring to this as a mode which my ET-15K didn't have. But it might be referring to the "Economic Mode" which can be configured on your inverter. However, in what form these have been implemented I have no idea and would love some more info.

Some insight could be provided if there were a known list of the parameters one could extract using the built in "Get parameter"-action/service.

**Comment by dmshimself** [comment]
I'm having trouble finding out what echo charge and discharge mode mean too.  Does anyone happen to know?

Similarly does fast charging switch and fast charging SOC only apply to those inverters equipped with an EV charging module or do those entities do something else?

**Reply by lampra1** [reply]
Eco charge : Charge from the grid  @ eco mode power
Eco discharge : Discharge to the grid  @ eco mode power
@ eco mode power = 0 both emulate General mode

**Reply by dmshimself** [reply]
Many thanks
